### ISSUE TITLE
Improve project_gdf performance

### ISSUE THREAD

**Issue by cheginit** [gboeing/osmnx#741]
**Is your feature proposal related to a problem?**

No.

**Describe the solution you'd like to propose**

For UTM projection in [`project_gdf`](https://github.com/gboeing/osmnx/blob/cd404ae2f153a5918a924b033e0e30700f156f99/osmnx/projection.py#L109) function, `unary_union` is being used which is a computationally expensive operation. Since for finding UTM there's no need for exact location of the centroid, I propose to use [`representative_point`](https://geopandas.org/docs/reference/api/geopandas.GeoSeries.representative_point.html#geopandas.GeoSeries.representative_point) function of GeoPandas to speed up the process. This is the function description:
> Returns a GeoSeries of (cheaply computed) points that are guaranteed to be within each geometry.

**Additional context**

This is the change that I am proposing:

Replace:
```python
avg_lng = gdf["geometry"].unary_union.centroid.x
```

with:
```python
avg_lng = gdf["geometry"].representative_point().x.mean()
```

An example for comparing their performance:

```python
import osmnx as ox

street = ox.graph_from_bbox(west=-96.0255, south=29.421, east=-94.782, north=30.512)
street = ox.graph_to_gdfs(
    street,
    nodes=False,
    node_geometry=False,
    fill_edge_geometry=True,
)
```

![image](https://user-images.githubusercontent.com/13016644/128642173-947ad2d7-bb4b-446b-8e5b-5b1baf08b783.png)

We can clearly see the performance gain from this change.

**Comment by gboeing** [comment]
This is a clever suggestion. Yes, I think `representative_point` is good enough for auto-determining a UTM zone. Would you like to open a simple PR to contribute this performance enhancement?

**Comment by cheginit** [comment]
Sure, would be happy to.

**Comment by cheginit** [comment]
Also, I found [this](https://github.com/Turbo87/utm/blob/40eb34c86895bf3a5f97b5819b9da4b164151d3c/utm/conversion.py#L301-L323) for computing zone number, do you want to follow that logic or prefer to keep it as is?

**Comment by gboeing** [comment]
> do you want to follow that logic or prefer to keep it as is?

Let's keep it simple and consistent for now. This might be worth revisiting in the future.

**Comment by gboeing** [comment]
Closed by #743.
